internal/transports/p2p/network/address_book.go

```diff
@@ -0,0 +1,128 @@
+package network
+
+import (
+	"fmt"
+	"math/rand"
+	"sync"
+	"time"
+
+	"github.com/bitcoin-sv/block-headers-service/internal/transports/p2p/peer"
+	"github.com/bitcoin-sv/block-headers-service/internal/wire"
+)
+
+type AddressBook struct {
+	banDuration time.Duration
+	addrs       []*knownAddress
+	keyIndex    map[string]int
+
+	mu           sync.Mutex
+	addrFitlerFn func(*wire.NetAddress) bool
+}
+
+func NewAdressbook(banDuration time.Duration, acceptLocalAddresses bool) *AddressBook {
+	addrFitlerFn := IsRoutable
+	if acceptLocalAddresses {
+		addrFitlerFn = IsRoutableWithLocal
+	}
+
+	return &AddressBook{
+		addrs:        make([]*knownAddress, 0, 500),
```

Comment:
### Minor 
I know that you won't like it, but maybe at least make 500 a const in the scope of this file? 🙈

Reply:
IMO it's unnecessary but moved to const